byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -26,23 +42,41 @@
 /** Servlet that handles data for stops */
 @WebServlet("/api/stop")
 public final class StopsServlet extends HttpServlet {
-  @GuardedBy("this")
-  private ArrayList<String> stops= new ArrayList<String>();
   private final Gson gson = new Gson(); 
-  // Implement datastore in routepage branch
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private static final Type ARRAYLIST_STRING = new TypeToken<ArrayList<String>>() {}.getType();
 
+  /* Passes saved destinations stops (if any) to be shown in the schedule panel */
   @Override
-  public synchronized void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Gson gson = new Gson();
-    String json = gson.toJson(stops);
-    
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: Replace ID filtering with .get(Key)
+    // Duplicate code was just kept for this reason
+    Filter propertyFilter = new FilterPredicate("id", FilterOperator.EQUAL, 1234);
+    Query query = new Query(Trip.KIND).setFilter(propertyFilter);
+    PreparedQuery results = datastore.prepare(query);
+    List<String> stops= new ArrayList<>();
```

Comment:
Change List to ArrayList

Reply:
I think both implementations are fine, because ArrayLists are just a specific implementation of the List interface.